local-volume/provisioner/pkg/deleter/proctable.go

```diff
@@ -0,0 +1,125 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package deleter
+
+import (
+	"fmt"
+	"sync"
+	"time"
+)
+
+// ProcTable Interface for tracking running processes
+type ProcTable interface {
+	// CleanupBlockPV deletes block based PV
+	IsRunning(pvName string) bool
+	IsEmpty() bool
+	MarkRunning(pvName string) error
+	MarkDone(pvName string)
+}
+
+var _ ProcTable = &ProcTableImpl{}
+
+// ProcEntry represents an entry in the proc table
+type ProcEntry struct {
```

Comment:
This can be done as a future task, but can we add metrics support so that we can get stats on how long deletion takes?

Reply:
OK, will do. Assuming this will be reported out via gRPC in minions, I will introduce it once the gRPC layer is in place.